agent-inject/agent/annotations.go

```diff
@@ -862,6 +862,14 @@ func (a *Agent) telemetryConfig() map[string]interface{} {
 func (a *Agent) authConfig() map[string]interface{} {
 	authConfig := make(map[string]interface{})
 
+	if a.Vault.Role != "" {
+		authConfig["role"] = a.Vault.Role
+	}
```

Comment:
I moved this to before the auth-config annotations loop because it would also be overridden if someone where to set the `vault.hashicorp.com/auth-config-role` annotation.